app/serializers/serialize_exercise_assignment.rb

```diff
@@ -87,7 +87,7 @@ def instructions_doc
 
   memoize
   def hints_doc
-    Markdown::Render.(exercise.git.hints, :doc)
+    Markdown::Render.(solution.git_exercise.hints, :doc)
```

Comment:
I think this should be cached in a `git_exercise` method so we don't init everything 3 times.

Reply:
It's a memoized method:

```
memoize
  def git_exercise
    Git::Exercise.for_solution(self)
  end
```